plugins/janus_streaming.c

```diff
@@ -6433,7 +6433,16 @@ static int janus_streaming_rtsp_connect_to_server(janus_streaming_mountpoint *mp
 	int asport = 0, asport_rtcp = 0;
 	multiple_fds audio_fds = {-1, -1};
 
-	janus_mutex_lock(&mountpoints_mutex);
+	if(g_atomic_int_get(&mp->destroyed)){
+		return -8;
+	}
+	
+	janus_mutex_lock(&mp->mutex);
+	
+	if(g_atomic_int_get(&mp->destroyed)){
+		janus_mutex_unlock(&mp->mutex);
+		return -8;
+	}
```

Comment:
Why check the same atomic int twice?

Reply:
since make sure the thread will break in time, rather than wait 5 seconds.
so...

Reply:
I still think checking twice makes no sense. Remove the first check, then, and only keep the one within the lock.